### PR TITLE
Update PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by jeffpaul**
### Description of the Change

This PR reorders some of the PR template sections, collapses some of the sections together, removes some of the sections/text, attempts to shorten the comments so they might regularly be read, and in general tries to simplify the PR template in hopes we actually get data from community contributions.

### Alternate Designs

Keep as-is with the longer format, but risk continuing to not get the most critical info we need within a PR description field.

### Possible Drawbacks

none identified

### Verification Process

n/a

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/10up/.github/blob/trunk/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Changelog Entry

n/a

### Credits

n/a

### CODE REVIEW

**Comment by jeffpaul** [comment]
@dkotter @dinhtungdu pinging you both on this as I've discussed with each of you in trying to simplify the PR template and hoping you can take a critical review here to see what else we might be able to simplify... thanks!

**Comment by fabiankaegy** [comment]
Just adding my 2cents here: 

I've really enjoyed doing Reviews of PRs that were made using the new [Pull Request Template of the Gutenberg Repository](https://github.com/WordPress/gutenberg/blob/trunk/.github/PULL_REQUEST_TEMPLATE.md). 

They split it up into three essential questions:
- What?
- Why?
- Testing instructions

And the thing that I have found to be the most helpful for me in reviewing a PR is actually the "Why" section because that often gets left out of regular PR descriptions. 

I personally am not the biggest fan of single-word questions because they seem a little harsh to me. But the message and idea behind that resonates with me. 

Maybe something like:
- Description of the Change
- Reason for the Change
- etc...

Would also achieve the same goal :)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,65 +1,42 @@
...
-
-### Verification Process
-
 <!--
-What process did you follow to verify that your change has the desired effects?
-
-- How did you verify that all new functionality works as expected?
-- How did you verify that all changed functionality works as expected?
-- How did you verify that the change has not introduced any regressions?
+We must be able to understand the design of your change from this description.  The maintainer reviewing this PR may not have worked with this code recently, so please provide as much deatil as possible.
```
Comment (dkotter) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
We must be able to understand the design of your change from this description.  The maintainer reviewing this PR may not have worked with this code recently, so please provide as much detail as possible.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,65 +1,42 @@
...
-
-<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
-<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
-- [ ] I have read the [**CONTRIBUTING**](https://github.com/10up/.github/blob/trunk/CONTRIBUTING.md) document.
-- [ ] My code follows the code style of this project.
-- [ ] My change requires a change to the documentation.
-- [ ] I have updated the documentation accordingly.
-- [ ] I have added tests to cover my change.
-- [ ] All new and existing tests passed.
+<!-- Enter any applicable Issue number(s) here that will be closed/resolved by this PR.-->
```
Comment (dkotter) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
<!-- Enter any applicable Issue number(s) here that will be closed/resolved by this PR. -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,65 +1,42 @@
...
+Closes #
 
-<!-- _NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open._ -->
 
 ### Changelog Entry
-
 <!--
-Add sample CHANGELOG.md entry for this PR, noting whether this is something being Added / Changed / Deprecated / Removed / Fixed / or Security related.
-Example:
+Please include a summary for this PR, noting whether this is something being Added / Changed / Deprecated / Removed / Fixed / or Security related.  You can replace the sample entries after this comment block with the single changelog entry line for this PR.-->
```
Comment (dkotter) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
Please include a summary for this PR, noting whether this is something being Added / Changed / Deprecated / Removed / Fixed / or Security related.  You can replace the sample entries after this comment block with the single changelog entry line for this PR. -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,65 +1,42 @@
...
-What process did you follow to verify that your change has the desired effects?
-
-- How did you verify that all new functionality works as expected?
-- How did you verify that all changed functionality works as expected?
-- How did you verify that the change has not introduced any regressions?
+We must be able to understand the design of your change from this description.  The maintainer reviewing this PR may not have worked with this code recently, so please provide as much deatil as possible.
 
-Describe the actions you performed (e.g., commands you ran, text you typed, buttons you clicked) and describe the results you observed.
+Where possible, please also include:
+- verification steps to ensure your change has the desired effects and has not introduced any regressions
```
Comment (dkotter) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: I like getting rid of the `Alternate Designs` and `Possible Drawbacks` sections, as those always confused me a little.

I may have been using it wrong but the `Verification Process` section I typically would use to put test instructions. May be worth adding a `How to test changes` section

**Comment by dkotter** [comment]
> I've really enjoyed doing Reviews of PRs that were made using the new [Pull Request Template of the Gutenberg Repository](https://github.com/WordPress/gutenberg/blob/trunk/.github/PULL_REQUEST_TEMPLATE.md).
> 
> I personally am not the biggest fan of single-word questions because they seem a little harsh to me. But the message and idea behind that resonates with me.

Yeah, I like the thought of what they're doing but it does confuse me at first glance of what they want in each of those sections (though the [code comments](https://raw.githubusercontent.com/WordPress/gutenberg/trunk/.github/PULL_REQUEST_TEMPLATE.md) help answer that).

We already have a `Description of the Change` section, which I would relate that directly to the `What` section.

For the `Why` section, we want all PRs to have an associated issue that explains the problem and ideally the PR would be linked to that issue using the `Closes` section, so I don't think we need another section for this.

For the `How` section, I guess I would expect to see whatever information is added there to also be in our `Description of the Change` section, but we could split those out if we wanted, where we have one section to briefly describe what is being fixed and another section that goes into more of the technical details? But if we're looking to simplify, I like just combing those into one.

And for the last, `Testing Instructions`, that is definitely a section I would like to see us add. I was always using our existing `Verification Process` section for this, though that's probably not what it was intended for.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (dinhtungdu) [reply .github/PULL_REQUEST_TEMPLATE.md:10]: I agree that we should have a dedicated section for testing instructions. Having a testing section will encourage contributors to provide detailed testing steps.

> I like getting rid of the `Alternate Designs` and `Possible Drawbacks` sections, as those always confused me a little.

I don't use these sections too 😅.

**Comment by jeffpaul** [comment]
I appreciate the super simplicity of the Gutenberg PR template, but I feel like it could make it's one-word questions more of a sentence approach to give a bit more context without having to read the full comments afterwards.  I do feel like that simplicity likely opens up for a broad range of info from contributors and potentially not what we at 10up are in search of as we review PRs.  All that said, I've added a testing section as I think being explicit on that is good (and is how I used the verification section previously).

@dkotter @dinhtungdu @fabiankaegy any further edits you think are worthwhile here before merging in and making use of in 10up's projects?

**Review by dkotter — APPROVED** [review]

**Pull request merged**